### PR TITLE
fix(product form): visually hide options rather than remove them if t…

### PR DESCRIPTION
**Pull request by stevenbister**
…here is only 1

### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [x] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [ ] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes https://www.notion.so/openstudycollege/Course-costs-not-pulling-through-from-Shopify-to-Sanity-504631980b454d64b3b21d0c42dccdda?pvs=4

## 📝 Description

Prices for variants with only one study method/course option is not updating the price when the other is selected in the product form

image
image

## ⛳️ Current behaviour (updates)

Currently when there is only one option available the product form will not render the options. This is leading to a bug where the url params aren't including the correct options to match against the selected variant.

## 🚀 New behaviour

Rather than removing the option completely I have updated it to pass a hidden class, so the option still exists in the form but is just not visible to the user.

## 💣 Is this a breaking change (Yes/No): No

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible cross-device impacts (iOS, Android)
- [x] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

### CODE REVIEW

File: packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx [hunk 1]
```diff
@@ -134,22 +134,21 @@ export const ProductForm = forwardRef<ElementRef<'div'>, ProductFormProps>(
...
-                                        }-${searchParamsWithDefaults.get(option.name)}`}
-                                        id={id}
-                                        option={option}
-                                        searchParamsWithDefaults={searchParamsWithDefaults}
-                                    />
-                                );
-                            }
-
-                            return <></>;
+                            // if (option.values.length >= 1) {
```
Comment (hi-im-si) [inline packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:137]: remove commented code

File: packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx [hunk 2]
```diff
@@ -134,22 +134,21 @@ export const ProductForm = forwardRef<ElementRef<'div'>, ProductFormProps>(
...
+                                    key={`${id}-${index}-${
+                                        option.name
+                                    }-${searchParamsWithDefaults.get(option.name)}`}
+                                    id={id}
+                                    option={option}
+                                    searchParamsWithDefaults={searchParamsWithDefaults}
+                                    className={option.values.length > 1 ? '' : 'u-hidden-from@all'}
+                                />
+                            );
+                            // }
```
Comment (hi-im-si) [inline packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:151]: remove commented code

File: packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx [hunk 3]
```diff
@@ -134,22 +134,21 @@ export const ProductForm = forwardRef<ElementRef<'div'>, ProductFormProps>(
...
+                                <Options
+                                    // Here we're passing the searchParamsWithDefaults as part of the key so the options remount when searchParams changes
+                                    // This is needed to keep the main form and the drawer forms in sync
+                                    key={`${id}-${index}-${
+                                        option.name
+                                    }-${searchParamsWithDefaults.get(option.name)}`}
+                                    id={id}
+                                    option={option}
+                                    searchParamsWithDefaults={searchParamsWithDefaults}
+                                    className={option.values.length > 1 ? '' : 'u-hidden-from@all'}
```
Comment (hi-im-si) [inline packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:148]: i would imagine we would not need to create a class for this and simply apply the `hidden` attribute?
```suggestion
                                    hidden={option.values.length <= 1 ? true : undefined}
                                    
```

**Review by hi-im-si — CHANGES_REQUESTED** [review]

File: packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx (hunk 2 above)
Comment (stevenbister) [reply packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:151]: Oops!

File: packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx (hunk 3 above)
Comment (stevenbister) [reply packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:148]: Yeah, that was my initial approach. But the inner component `RadioGroup` doesn't extend the html attributes for the `fieldset` `hidden` didn't work :/ I guess I could wrap the `Options` in a div and put it there but not keen on the extra markup for the sake of a `hidden` property :/

Comment (hi-im-si) [reply packages/osc-ecommerce/app/components/Forms/ProductForm/ProductForm.tsx:161]: ahh no worries makes sense.

**Comment by stevenbister** [comment]
@hi-im-si Noticed I missed a bit of logic, so would still fail to update the cost if there is only one study pack e.g http://localhost:2000/courses/aat-accounting-level-2-and-3?Course+Options=Course+Material&Study+Options=Study+Pack. I've updated the condition to render the `Options` there too

**Pull request merged**